community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/Neo4jTransactionContext.scala

```diff
@@ -17,24 +17,35 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-package org.neo4j.cypher.internal.spi.v3_0
+package org.neo4j.cypher.internal
 
-import org.neo4j.cypher.internal.compiler.v3_0.spi.TransactionalContext
-import org.neo4j.graphdb.Transaction
+import org.neo4j.cypher.internal.spi.ExtendedTransactionalContext
+import org.neo4j.graphdb.{Lock, PropertyContainer, Transaction}
 import org.neo4j.kernel.GraphDatabaseQueryService
-import org.neo4j.kernel.api.Statement
+import org.neo4j.kernel.api.{ReadOperations, Statement}
 import org.neo4j.kernel.api.txstate.TxStateHolder
 import org.neo4j.kernel.impl.api.KernelStatement
 import org.neo4j.kernel.impl.core.ThreadToStatementContextBridge
 
-case class TransactionBoundTransactionalContext(graph: GraphDatabaseQueryService, initialTx: Transaction,
-                                                val isTopLevelTx: Boolean, initialStatement: Statement)
-  extends TransactionalContext[GraphDatabaseQueryService, Statement, TxStateHolder] {
+object TransactionContextFactory {
+  def open(graphDatabaseAPI: GraphDatabaseQueryService, txBridge: ThreadToStatementContextBridge): Neo4jTransactionContext = {
+    val isTopLevelTx = !txBridge.hasTransaction
+    val tx = graphDatabaseAPI.beginTx()
+    val statement = txBridge.get()
+    Neo4jTransactionContext(graphDatabaseAPI, tx, isTopLevelTx, statement)
+  }
+}
+
+// please construct this class through TransactionContextFactory, this is public only for tests
```

Comment:
I feel like this comment is bound to be overlooked at some point. Is there another way to design this, and still allow for testing? A test subclass?